packages/react-reconciler/src/ReactFiberBeginWork.new.js

```diff
@@ -972,7 +972,10 @@ function updateTracingMarkerComponent(
   // TODO: (luna) Only update the tracing marker if it's newly rendered or it's name changed.
   // A tracing marker is only associated with the transitions that rendered
   // or updated it, so we can create a new set of transitions each time
-  if (current === null) {
+  if (
+    current === null ||
+    current.memoizedProps.name === workInProgress.pendingProps.name
```

Comment:
Why did you move this?

Reply:
I didn't exactly move it. The previous check was `current.memoizedProps.name !== workInProgress.pendingProps.name`. But if the name is the same and the component rerenders we should update it (because most likely it was unhidden)